packages/modular-scripts/src/test.ts

```diff
@@ -1,70 +1,71 @@
 import * as path from 'path';
 import resolve from 'resolve';
-import getModularRoot from './utils/getModularRoot';
+import resolveBin from 'resolve-as-bin';
 import execSync from './utils/execSync';
-import resolveAsBin from 'resolve-as-bin';
-import createJestConfig from './config/jest';
+import getModularRoot from './utils/getModularRoot';
+export interface TestOptions {
+  debug: boolean;
+  coverage: boolean;
+  forceExit: boolean;
+  env: string;
+  maxWorkers: number;
+  onlyChanged: boolean;
+  json: boolean;
+  outputFile: string;
+  reporters: string[] | undefined;
+  runInBand: boolean;
+  silent: boolean;
+  testResultsProcessor: string | undefined;
+  updateSnapshot: boolean;
+  verbose: boolean;
+  watch: boolean;
+  watchAll: boolean;
+}
 
-type VerifyPackageTree = () => void;
+// via https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/test.js
+// This is a very dirty workaround for https://github.com/facebook/jest/issues/5913.
+// We're trying to resolve the environment ourselves because Jest does it incorrectly.
+// TODO: remove this as soon as it's fixed in Jest.
+function resolveJestDefaultEnvironment(name: string) {
+  const jestDir = path.dirname(
+    resolve.sync('jest', {
+      basedir: __dirname,
+    }),
+  );
+  const jestCLIDir = path.dirname(
+    resolve.sync('jest-cli', {
+      basedir: jestDir,
+    }),
+  );
+  const jestConfigDir = path.dirname(
+    resolve.sync('jest-config', {
+      basedir: jestCLIDir,
+    }),
+  );
+  return resolve.sync(name, {
+    basedir: jestConfigDir,
+  });
+}
 
-export default function test(args: string[]): Promise<void> {
-  if (process.env.SKIP_PREFLIGHT_CHECK !== 'true') {
-    const verifyPackageTree =
-      require('react-scripts/scripts/utils/verifyPackageTree') as VerifyPackageTree; // eslint-disable-line @typescript-eslint/no-var-requires
-    verifyPackageTree();
-  }
+export default async function test(
+  options: TestOptions,
+  regexes?: string[],
+): Promise<void> {
+  const { debug, env, reporters, testResultsProcessor, ...jestOptions } =
+    options;
 
-  const modularRoot = getModularRoot();
-  const jestConfig = createJestConfig();
-  let argv = process.argv
-    .slice(3)
-    .concat(['--config', JSON.stringify(jestConfig)]);
+  // create argv from jest Options
+  const cleanArgv: string[] = [];
 
-  // Watch unless on CI or explicitly running all tests
-  if (!process.env.CI && args.indexOf('--watchAll=false') === -1) {
-    // https://github.com/facebook/create-react-app/issues/5210
-    argv.push('--watchAll');
-  }
+  // pass in path to configuration file
+  const { createJestConfig } = await import('./config/jest');
```

Comment:
Why do we need to wait to import this function?

Reply:
I wanted to hide away the jest imports given I know it has a deep dependency tree - so this should minimize the import cost of test.